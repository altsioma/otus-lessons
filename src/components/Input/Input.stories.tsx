import React, { useState } from "react";
import { Input } from "../../components/Input/Input";
import { action } from "@storybook/addon-actions";
import { withKnobs } from "@storybook/addon-knobs";

export default {
  title: "GraphMakerComponents",
  decorators: [withKnobs],
};
const inputChange = action("Input change");

export const InputControl: React.FC<{}> = () => {
  const [value, setValue] = useState<string>();
  const onChange = (e: React.ChangeEvent) => {
    inputChange((e.target as HTMLInputElement).value);
    setValue((e.target as HTMLInputElement).value);
  };
  return (
    <Input
      labelText="Label:"
      type="text"
      placeholder="Placeholder"
      value={value}
      onChange={onChange}
    />
  );
};
