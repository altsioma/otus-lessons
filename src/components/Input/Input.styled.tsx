import React from "react";
import styled from "@emotion/styled";
import { css } from "@emotion/core";

const BaseInput = css`
margin: 10px 0;
display: flex;
justify-content: flex-end;
`;

export const LabelStyled = styled.label`
  font-size: 16px;
  margin-bottom: 5px;
  padding: 1em 1em 0.5em 0;
  width: 7%;
`;

export const InputStyled = styled.input`
  height: 40px;
  width: 100%;
  border: 1px solid rgba(0, 0, 0, 0.2);
  border-radius: 5px;
  font-size: 18px;
  padding: 5px 10px;
  flex: 1;
`;

export const InputWrapperStyled = styled.div`
  ${BaseInput};
`;
