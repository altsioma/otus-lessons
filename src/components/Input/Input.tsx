import React from "react";
import styled from "@emotion/styled";
import { css } from "@emotion/core";

const Label = styled.label`
  font-size: 16px;
  margin-bottom: 5px;
  padding: .5em 1em .5em 0;
  width: 10%;
`;

const InputStyled = styled.input`
  height: 40px;
  width: 100%;
  border: 1px solid rgba(0, 0, 0, 0.2);
  border-radius: 5px;
  font-size: 18px;
  padding: 5px 10px;
  flex: 1;
`;

const BaseInput = css`
  margin: 10px 0;
  display: flex;
  justify-content: flex-end;
`;

const InputWrapper = styled.div`
  ${BaseInput};
`;

interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  labelText: string;
}

export const Input: React.FC<InputProps> = ({ labelText, ...props }) => {
  return (
  <InputWrapper>
    <Label>
      {labelText}
    </Label>
    <InputStyled {...props} />
  </InputWrapper>
  )
};
