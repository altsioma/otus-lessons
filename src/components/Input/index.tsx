import React from "react";
import { LineType } from "../../types"
// import "./index.css";

type InputType = {
  value: LineType;
  onChnage: () => void;
};

export class Input extends React.Component<InputType> {
  constructor(props: InputType) {
    super(props);
  }

  render() {
    return (
      <div>
        <span>X=</span>
        <input value={this.props.value[0]} />
        <span>Y=</span>
        <input value={this.props.value[1]} />
      </div>
    );
  }
}
