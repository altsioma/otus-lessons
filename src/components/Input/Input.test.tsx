import React from "react";
import { Input } from "./Input";
import renderer from "react-test-renderer";

describe("Input", () => {
  it("renders Input with value", () => {
    expect(
      renderer
        .create(
          <Input
            labelText="Label:"
            type="text"
            placeholder="Placeholder"
            value={"Testing"}
            onChange={jest.fn()}
          />
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});
