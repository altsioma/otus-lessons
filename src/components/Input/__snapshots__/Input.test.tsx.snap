// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders Input with value 1`] = `
<div
  className="css-1wbj7b9"
>
  <label
    className="css-eq8zle"
  >
    Label:
  </label>
  <input
    className="css-5mhwm8"
    onChange={[MockFunction]}
    placeholder="Placeholder"
    type="text"
    value="Testing"
  />
</div>
`;
